bigtable/integration_test.go

```diff
@@ -899,6 +915,51 @@ func TestIntegration_LargeReadsWritesAndScans(t *testing.T) {
 	} else if status[0] == nil || status[1] == nil {
 		t.Fatalf("No error for bad bulk mutation")
 	}
+
+	// Validate that metrics are exported
+	elapsedTime := time.Since(testStartTime)
+	if elapsedTime < 2*defaultSamplePeriod {
+		// Ensure at least 2 datapoints are recorded
+		time.Sleep(2*defaultSamplePeriod - elapsedTime)
+	}
+
+	timeListEnd := time.Now()
+	tsListEnd := &timestamppb.Timestamp{
+		Seconds: timeListEnd.Unix(),
+		Nanos:   int32(timeListEnd.Nanosecond()),
+	}
+
+	monitoringClient, err := monitoring.NewMetricClient(ctx)
+	if err != nil {
+		t.Errorf("Failed to create metric client: %v", err)
+	}
+
+	metricNamesValidate := []string{
+		metricNameOperationLatencies,
+		metricNameAttemptLatencies,
+		metricNameServerLatencies,
+	}
+	for _, metricName := range metricNamesValidate {
+		// ListTimeSeries can list only one metric type at a time.
+		// So, call ListTimeSeries with different metric names
+		iter := monitoringClient.ListTimeSeries(ctx, &monitoringpb.ListTimeSeriesRequest{
+			Name: fmt.Sprintf("projects/%s", testEnv.Config().Project),
+			Interval: &monitoringpb.TimeInterval{
+				StartTime: tsListStart,
+				EndTime:   tsListEnd,
+			},
+			Filter: fmt.Sprintf("metric.type = starts_with(\"bigtable.googleapis.com/client/%v\")", metricName),
+		})
+
+		// Assert at least 1 datapoint was exported
+		_, err := iter.Next()
+		if testEnv.Config().UseProd && err != nil {
+			t.Errorf("%v not exported\n", metricName)
+		}
+		if !testEnv.Config().UseProd && err != iterator.Done {
+			t.Errorf("%v should not be exported when using emulator\n", metricName)
+		}
+	}
```

Comment:
I would skip this test entirely for the emulator can just use a simple unit test to ensure that the default metrics provider is the noop one

Reply:
Done